shotover-proxy/src/transforms/cassandra/connection.rs

```diff
@@ -26,33 +27,37 @@ struct Request {
 
 #[derive(Clone, Derivative)]
 #[derivative(Debug)]
-pub struct CassandraConnection<C: Codec> {
+pub struct CassandraConnection {
     host: String,
-    connection: Option<mpsc::UnboundedSender<Request>>,
-    #[derivative(Debug = "ignore")]
-    codec: C,
+    connection: mpsc::UnboundedSender<Request>,
 }
 
-impl<C: Codec + 'static> CassandraConnection<C> {
-    pub fn new(host: String, codec: C) -> Self {
-        CassandraConnection {
-            host,
-            connection: None,
-            codec,
-        }
-    }
+impl CassandraConnection {
+    pub async fn new<C: Codec + 'static>(
+        host: String,
+        codec: C,
+        mut tls: Option<TlsConnector>,
+    ) -> Result<Self> {
+        let tcp_stream: TcpStream = TcpStream::connect(&host).await?;
 
-    pub async fn connect(&mut self) -> Result<()> {
-        let socket: TcpStream = TcpStream::connect(self.host.clone()).await?;
-        let (read, write) = socket.into_split();
         let (out_tx, out_rx) = mpsc::unbounded_channel::<Request>();
         let (return_tx, return_rx) = mpsc::unbounded_channel::<Request>();
 
-        tokio::spawn(tx_process(write, out_rx, return_tx, self.codec.clone()).in_current_span());
+        if let Some(tls) = tls.as_mut() {
+            let tls_stream = tls.connect(tcp_stream).await.unwrap();
```

Comment:
Should we return Err instead of unwrap here?
Currently we never return Err in this function despite being Result<()>

Lets make the redis equivalent call do the same as well.